src/ui/pages/AssetInfo/AssetResources.tsx

```diff
@@ -0,0 +1,365 @@
+              />
+              <VStack gap={0}>
+                <UIText kind="body/accent">{network.name}</UIText>
+                {address ? (
+                  <UIText kind="small/regular" color="var(--neutral-500)">
+                    {middleTruncate({ value: address })}
+                  </UIText>
+                ) : null}
+              </VStack>
+            </HStack>
+            <HStack gap={0}>
+              {address ? <CopyAddressButton address={address} /> : null}
+              <UnstyledAnchor
+                rel="noopener noreferrer"
+                target="_blank"
+                href={
+                  address && networks
+                    ? networks.getExplorerTokenUrlByName(
+                        createChain(network.id),
+                        address
+                      )
+                    : network.explorer_home_url || ''
+                }
+                className="parent-hover"
+                style={{
+                  ['--parent-content-color' as string]: 'var(--neutral-400)',
+                  ['--parent-hovered-content-color' as string]:
+                    'var(--neutral-700)',
+                  display: 'flex',
+                }}
+              >
+                <LinkIcon className="content-hover" />
+              </UnstyledAnchor>
+            </HStack>
+          </HStack>
+        ))}
+      </VStack>
+    </VStack>
+  );
+}
+
+function ResourseButton({
+  resource,
+  icon,
+}: {
+  resource: AssetResource;
+  icon: React.ReactNode;
+}) {
+  return (
+    <Button
+      key={resource.name}
+      as={UnstyledAnchor}
+      rel="noopenner noreferrer"
+      target="_blank"
+      kind="regular"
+      size={36}
+      href={resource.url}
+      style={{
+        padding: 6,
+        border: '2px solid var(--neutral-200)',
+        ['--button-background' as string]: 'var(--white)',
+        ['--button-background-hover' as string]: 'var(--neutral-200)',
+      }}
+      aria-label={resource.displayableName}
+    >
+      {icon}
+    </Button>
+  );
+}
+
+const TWITTER_ID = 'twitter';
+const WARPCAST_ID = 'warpcast';
+const DEXSCREENER_ID = 'dexscreener';
+
+export function AssetResources({
+  assetFullInfo,
+}: {
+  assetFullInfo: AssetFullInfo;
+}) {
+  const { networks } = useNetworks();
+  const dialogRef = useRef<HTMLDialogElementInterface | null>(null);
+  const implementations = useMemo(
+    () =>
+      Object.entries(assetFullInfo.fungible.implementations)
+        .map(([networkId, config]) => {
+          const network = networks?.getNetworkByName(createChain(networkId));
+          return network ? { address: config.address, network } : null;
+        })
+        .filter(isTruthy),
+    [assetFullInfo.fungible.implementations, networks]
+  );
+
+  const mainImplementation = useMemo(() => {
+    return (
+      implementations.find(
+        (implementation) =>
+          implementation.network.id === assetFullInfo.extra.mainChain
+      ) || implementations[0]
```

Comment:
```suggestion
      ) || implementations.at(0)
```

this will type the value as optional